synse/http_integration_test.go

```diff
@@ -0,0 +1,551 @@
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	opts := scheme.ScanOptions{}
+	devices, err := client.Scan(opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 22, len(devices))
+
+	for _, device := range devices {
+		opts := scheme.ReadOptions{}
+		reads, err := client.ReadDevice(device.ID, opts)
+		assert.NoError(t, err)
+
+		for _, read := range reads {
+			assert.NotEmpty(t, read.Device)
+			assert.NotEmpty(t, read.Timestamp)
+			// assert.NotEmpty(t, read.Type) // FIXME - fan and airflow types are empty?
+			assert.NotEmpty(t, read.DeviceType)
+
+			// led and lock devices don't produce unit output
+			if read.DeviceType == "led" || read.DeviceType == "lock" {
+				assert.Empty(t, read.Unit.Name)
+				assert.Empty(t, read.Unit.Symbol)
+			} else {
+				assert.NotEmpty(t, read.Unit.Name)
+				assert.NotEmpty(t, read.Unit.Symbol)
+			}
+
+			// NOTE - read.Value could be 0 so no need to check that
+		}
+	}
+}
+
+func TestIntegration_ReadCache(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	opts := scheme.ReadCacheOptions{}
+	reads := make(chan *scheme.Read, 1)
+
+	go func() {
+		err := client.ReadCache(opts, reads)
+		assert.NoError(t, err)
+	}()
+
+	for {
+		var done bool
+		select {
+		case read, open := <-reads:
+			if !open {
+				done = true
+				break
+			}
+
+			assert.NotEmpty(t, read.Device)
+			assert.NotEmpty(t, read.Timestamp)
+			// assert.NotEmpty(t, read.Type) // FIXME - fan and airflow types are empty?
+			assert.NotEmpty(t, read.DeviceType)
+
+			// led and lock devices don't produce unit output
+			if read.DeviceType == "led" || read.DeviceType == "lock" {
+				assert.Empty(t, read.Unit.Name)
+				assert.Empty(t, read.Unit.Symbol)
+			} else {
+				assert.NotEmpty(t, read.Unit.Name)
+				assert.NotEmpty(t, read.Unit.Symbol)
+			}
+
+			// NOTE - read.Value could be 0 so no need to check that
+
+		case <-time.After(2 * time.Second):
+			// if the test does not complete after 2s, error.
+			t.Fatal("timeout: failed getting readcache data from channel")
+		}
+
+		if done {
+			break
+		}
+	}
+}
+
+func TestIntegration_WriteAsync(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	// collect all writable devices.
+	// FIXME - refer to #69, can only query one type atm.
+	opts := scheme.ScanOptions{
+		Tags: []string{
+			"system/type:fan",
+			// "system/type:led",
+			// "system/type:lock",
+			// "system/type:power",
+		},
+	}
+	devices, err := client.Scan(opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(devices))
+
+	for _, device := range devices {
```

Comment:
I feel like instead of scanning for devices and iterating over a slice of 1, we can just use a well-known device ID for the write. This gets to the point above about defining a custom device configuration for these tests. We could also define aliases for devices so instead of using some opaque hash string, we could reference devices by a more readable name.